Source/missiles.cpp

```diff
@@ -590,6 +590,90 @@ void CheckMissileCol(Missile &missile, int mindam, int maxdam, bool shift, Point
 		PlaySfxLoc(MissilesData[missile._mitype].miSFX, missile.position.tile);
 }
 
+/**
+ * @brief Could the missile collide with solid objects? (like walls or closed doors)
+ */
+bool CouldMissileCollideWithSolidObject(Point tile)
+{
+	int oid = dObject[tile.x][tile.y];
+	if (oid != 0) {
+		oid = oid > 0 ? oid - 1 : -(oid + 1);
+		if (!Objects[oid]._oMissFlag)
+			return true;
+	}
+	return nMissileTable[dPiece[tile.x][tile.y]];
+}
+
+void MoveMissileAndCheckMissileCol(Missile &missile, int mindam, int maxdam, bool ignoreStart, bool ifCollidesDontMoveToHitTile)
+{
+	Point prevTile = missile.position.tile;
+	missile.position.traveled += missile.position.velocity;
+	UpdateMissilePos(missile);
+	int walkingDistance = prevTile.WalkingDistance(missile.position.tile);
+
+	int16_t tileTargetIdentifer = dMonster[missile.position.tile.x][missile.position.tile.y] - dPlayer[missile.position.tile.x][missile.position.tile.y];
+
+	if (walkingDistance == 0) {
+		// missile didn't change the tile... check that we perform CheckMissileCol only once for any monster/player to avoid multiple hits for slow missiles
+		if (missile.lastCollisionTargetIdentifer == tileTargetIdentifer)
+			return;
+	}
+	// remember what target CheckMissileCol was checked against
+	missile.lastCollisionTargetIdentifer = tileTargetIdentifer;
+	// Did the missile skipped a tile?
+	if (walkingDistance > 1) {

```

Comment:
If a missile passes over an adjacent tile into a diagonal tile in a single bound, this would skip the collision test for the adjacent tile.

Reply:
Totally missed this one. 😄 
I tried to fix it with checking coordinates diff and only use `WalkingDistance` if the missile flies straight in one direction (only x or y).
Hope you can look again. 🙂